open_spiel/games/battleship_test.cc

```diff
@@ -0,0 +1,261 @@
+// Copyright 2020 DeepMind Technologies Ltd. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "open_spiel/games/battleship.h"
+
+#include <iostream>
+#include <limits>
+
+#include "open_spiel/algorithms/expected_returns.h"
+#include "open_spiel/algorithms/get_all_states.h"
+#include "open_spiel/algorithms/tabular_exploitability.h"
+#include "open_spiel/policy.h"
+#include "open_spiel/spiel_utils.h"
+#include "open_spiel/tests/basic_tests.h"
+
+namespace open_spiel {
+namespace battleship {
+namespace {
+
+namespace testing = open_spiel::testing;
+
+void BasicBattleshipTest() {
+  // Some basic tests on a small 2x2 instance.
+  for (int num_shots = 1; num_shots <= 3; ++num_shots) {
+    const auto game =
+        LoadGame("battleship", {{"board_width", GameParameter(2)},
+                                {"board_height", GameParameter(2)},
+                                {"ship_sizes", GameParameter("[1;2]")},
+                                {"ship_values", GameParameter("[1;2]")},
+                                {"num_shots", GameParameter(num_shots)},
+                                {"allow_repeated_shots", GameParameter(false)},
+                                {"loss_multiplier", GameParameter(2.0)}});
```

Comment:
Up to you, but if you want to, you can add some of these to `canonical_game_strings.h` if they're particularly notable: 

https://github.com/deepmind/open_spiel/blob/837b5430721197682fc232eac74291366a765054/open_spiel/canonical_game_strings.h

(You can build a game string up as `battleship(board_width=2, board_height=2,...)`)

Reply:
Thanks for the note! I think for now I'd rather not add a canonical game string. I'm sure there will be opportunities in the future (for example, when trying to come up with exact optimal algorithms for EFCE/EFCCE/NFCCE) to factor out small "canonical" battleship games for tests, depending on what we end up using.

Reply:
That would be awesome!